api/multicluster/v1alpha1/cluster.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+package multicluster.solo.io;
+
+import "gogoproto/gogo.proto";
+option (gogoproto.equal_all) = true;
+import "extproto/ext.proto";
+option (extproto.hash_all) = true;
+
+option go_package = "github.com/solo-io/skv2/pkg/api/multicluster.solo.io/v1alpha1/types";
+
+
+
+/*
+    Representation of a Kubernetes cluster that has been registered.
+*/
+message KubernetesClusterSpec {
+    // name of the secret which contains the kubeconfig with information to connect to the remote cluster.
+    string secret = 1;
```

Comment:
i'd like to allow the user to provide a secret_key here as well. we can override emtpy with a default

Reply:
What do you mean by a `secret_key`

Reply:
I think he means the key of the kubeconfig entry in the secret.

```yaml
kind: Secret
metadata:
  ...
data:
  <cluster.secret_key | "some-default">: encoded_kubeconfig
```